Telemetry - re-introduce telemetry.

sofar opened the pull request:
This is a new telemetry framework for swupd-client. The patch is split up in a few changes:

- clean out some old telemetry stuff that was left around and is unused.
- Separate patch to always create new telemetry drop folder
- introduce new telemetry() function and use it in 3 locations (check, update, and verify paths) to create a telemetry record at completion of those actions.

This won't actually deliver records to the telemetry service. For that, a separate telemetry agent will be created that runs as a separate process and does not affect swupd operation. That "probe" will have its own repository.

sofar commented:
Fixed up all the telemetry messages starting with `action=`.

tmarcu commented on src/helpers.c:
```diff
@@ -135,8 +135,8 @@ static int create_required_dirs(void)
-#define STATE_DIR_COUNT 3
-	const char *dirs[] = { "delta", "staged", "download" };
+#define STATE_DIR_COUNT 4
+	const char *dirs[] = { "delta", "staged", "download", "telemetry" };
```
This should maybe be renamed to state_dirs so it's obvious it's referencing the above define.

phmccarty replied:
That would be an improvement, though it could be incorporated in a separate cleanup patch.

sofar replied:
Done, separate commit 2nd in this series.

phmccarty commented on src/telemetry.c:
```diff
@@ -0,0 +1,61 @@
+
+	string_or_die(&filename, "%s/telemetry/%d.%d.XXXXXX", state_dir, RECORD_VERSION, level);
+
+	if (!filename) {
```
`filename` must be non-null at this point, because `string_or_die` aborts if memory allocation fails.

sofar replied:
ok, thanks, I'll remove this check.

phmccarty commented on src/update.c:
```diff
@@ -400,6 +404,22 @@ int main_update()
+		ret,
+		delta);
+
+	if (server_version > current_version)
```
This `if` block needs surrounding curly brackets (code style)

sofar replied:
yup!

phmccarty commented on src/telemetry.c:
```diff
@@ -0,0 +1,61 @@
+/*
+ *   Software Updater - telemetry collection
+ *
+ *      Copyright © 2012-2016 Intel Corporation.
```
just 2016?

sofar commented:
I've fixed up all the comments, and added 1 more commit to rename the `dirs` to `state_dirs`. I've inserted it before the series since that makes more sense.

phmccarty commented:
+1
